### DISCUSSION TITLE
Manual cache update after mutation error.

### DISCUSSION THREAD

**Discussion by augustas1** [urql-graphql/urql#2297] [Q&A] [answered]
I've been using manual graphcache updates to update existing queries.
But I realized if mutation fails, manual cache update would still be called.

In the example below, if mutation fails, it would write `null` to the cache (and wipe existing user data if any), because mutation would return `createUser: null`.

I know I could check `info.error` for failure, but I wouldn't want to do that each time.

Should manual cache updates not be triggered on mutation failure?

```
updates: {
  Mutation: {
    createUser(result, args, cache, info) {
      cache.updateQuery(
        { query: UserQuery, variables: { userId: args.userId } },
        () => ({ user: result.createUser })
      );
    }
  }
}
```

**Comment by JoviDeCroock** [comment, marked as answer]
Well, GraphQL can fail partially, what if your mutation succeeds but a sub-resolver for your selected entity fails? Your mutation executed so you fo want to update the data you are getting.

**Comment by govorukhin-do** [comment]
How you do this? My updaters doesn't  run when mutations fails, but I need update cache on fail
